pkg/apis/capabilities/v1beta1/openapi_types.go

```diff
@@ -0,0 +1,193 @@
+package v1beta1
+
+import (
+	"reflect"
+
+	"github.com/3scale/3scale-operator/pkg/common"
+
+	"github.com/go-logr/logr"
+	"github.com/google/go-cmp/cmp"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/validation/field"
+)
+
+// EDIT THIS FILE!  THIS IS SCAFFOLDING FOR YOU TO OWN!
+// NOTE: json tags are required.  Any new fields you add must have json tags for the fields to be serialized.
+
+const (
+	OpenapiKind = "Openapi"
+
+	// OpenapiInvalidConditionType represents that the combination of configuration
+	// in the OpenapiSpec is not supported. This is not a transient error, but
+	// indicates a state that must be fixed before progress can be made.
+	// Example: the spec references invalid openapi spec
+	OpenapiInvalidConditionType common.ConditionType = "Invalid"
+
+	// OpenapiReadyConditionType indicates the openapi resource has been successfully reconciled.
+	// Steady state
+	OpenapiReadyConditionType common.ConditionType = "Ready"
+
+	// OpenapiFailedConditionType indicates that an error occurred during reconcilliation.
+	// The operator will retry.
+	OpenapiFailedConditionType common.ConditionType = "Failed"
+)
+
+// OpenAPIRefSpec Reference to the OpenAPI Specification
+type OpenAPIRefSpec struct {
+	// ConfigMapRef ConfigMap that contains the OpenAPI Document
+	// +optional
+	ConfigMapRef *corev1.ObjectReference `json:"configMapRef,omitempty"`
```

Comment:
I wonder whether we should be using `ObjectReference`. Wouldn't a `LocalObjectReference` be enough?
`ObjectReference` has lots of fields that depend on the user creating them and forces us to unintuitive changes and checkings throughout our code (see setDefaults methods and potentially other checkings in the code)

Reply:
The idea was not to limit to the local namespace itself the location of the configmap. One customer may have a configmap in some other namespace and providing custom access to `3scale-operator` service account, the operator would be able to access config maps from other namespaces than local namespace. 

Maybe we should only define `Name` and `Namespace` attributes.